### PR TITLE
Add support for declaring contract compiled with cairo 2.6

### LINKED ISSUE #1315: Add compiled_class_hash for contract compiled with cairo 2.6.0

### PR DESCRIPTION
**Pull request by tkumor3**
Closes #1315

- Add support for declaring contract compiled with cairo 2.6

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: .github/workflows/checks.yml
Comment (THenry14) [inline .github/workflows/checks.yml]: why not use scarb?

File: .vscode/settings.json
Comment (THenry14) [inline .vscode/settings.json:1]: should this file be included?

File: .github/workflows/checks.yml
Comment (THenry14) [inline .github/workflows/checks.yml]: I think it may make sense to switch the v2 scenarios - we will probably stay at 2.5.X version of a compiler, and update 2.6 to newer versions from now on

File: starknet_py/hash/casm_class_hash_test.py
Comment (THenry14) [reply starknet_py/hash/casm_class_hash_test.py:37]: people will see this link and won't be able to access an info behind it.

File: starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh
Comment (THenry14) [inline starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh:11]: nitpick: in bash, use 2 spaces instead of 4 for indentation

File: starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh
Comment (THenry14) [inline starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh:1]: This script has several smaller issues. Did you copy it over from some place, or did you write it from scratch? (I'm wondering if we should just make an issue to fix those, or should this script be fixed in this PR)

Comment (tkumor3) [reply starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh:1]: IMO we should make an issue and change this scripts to use scarb

File: .github/workflows/checks.yml
Comment (tkumor3) [reply .github/workflows/checks.yml]: These changes are moved to the new PR https://github.com/software-mansion/starknet.py/pull/1322

File: starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh
Comment (tkumor3) [reply starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh:11]: These changes will be added in #1322

File: starknet_py/tests/e2e/declare/declare_test.py
Comment (ddoktorski) [inline starknet_py/tests/e2e/declare/declare_test.py:45]: As you've noticed this is not a very valuable test, given that in the devnet we can pass anything to `compiled_class_hash` and it will be successful. Nevertheless, I'm fine with leaving it, as perhaps it will be fixed at some point. However, instead of adding a new test, please make `test_declare_v3` a parameterized test for 2 contracts.

File: starknet_py/net/schemas/gateway.py
Comment (ddoktorski) [inline starknet_py/net/schemas/gateway.py:169]: nit:
```suggestion
    bytecode_segment_lengths = fields.List(Felt(), data_key="bytecode_segment_lengths", load_default=None)
```

File: starknet_py/tests/e2e/mock/contracts_compiled_v2_6/starknet_contract.casm
Comment (tkumor3) [reply starknet_py/tests/e2e/mock/contracts_compiled_v2_6/starknet_contract.casm:1]: This contract is taken from a conversation with Starknet on Slack. They sent only this contract file  and the correct class_hash for it

File: starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh
Comment (tkumor3) [reply starknet_py/tests/e2e/mock/compile_contracts_v2_6.sh:1]: It will be done in #1322. Probably we will migrate scripts to Scarb

File: starknet_py/tests/e2e/mock/contracts_v2_6/abi_types.cairo
Comment (tkumor3) [reply starknet_py/tests/e2e/mock/contracts_v2_6/abi_types.cairo:1]: it's cleaning up in #1322, so the mentioned PR aims is to compile contracts_v2 with the current Cairo version and run all tests with it.   Extended with precompiled contract tests to verify if class_hashs are computed well for all Sierra versions

File: starknet_py/tests/e2e/declare/declare_test.py
Comment (tkumor3) [reply starknet_py/tests/e2e/declare/declare_test.py:45]: test with `starknet_contract.casm` cover this test case

Comment (tkumor3) [reply starknet_py/tests/e2e/declare/declare_test.py:45]: I think we can remove this test and most of the things related to the 2.6 contract. The truth is that we need only `starknet_contract.casm`

**Review by THenry14 — APPROVED** [review]
looks good to me, but let's wait for @ddoktorski approval as well

File: starknet_py/hash/casm_class_hash.py
Comment (ddoktorski) [inline starknet_py/hash/casm_class_hash.py:37]: nit: This assert doesn't seem to be necessary.
```suggestion
```

**Review by ddoktorski — APPROVED** [review]

**Pull request merged**
